xbmc/Application.cpp

```diff
@@ -595,21 +595,18 @@ bool CApplication::Create(const CAppParamParser &params)
   m_pWinSystem = CWinSystemBase::CreateWinSystem();
   CServiceBroker::RegisterWinSystem(m_pWinSystem.get());
 
-  if (!m_ServiceManager->CreateAudioEngine())
-  {
-    CLog::Log(LOGFATAL, "CApplication::Create: Failed to load an AudioEngine");
-    return false;
-  }
+  m_pActiveAE.reset(new ActiveAE::CActiveAE());
+  CServiceBroker::RegisterAE(m_pActiveAE.get());
   // start the AudioEngine
-  if(!m_ServiceManager->StartAudioEngine())
+  if (!m_pActiveAE->Initialize())
```

Comment:
Should we register AE after initialization so clients can be guaranteed an initialized AE? We do the reverse on shutdown.

Reply:
yes. I will change it. thanks